---
title: "RNA-seq"
author: "Jingyi Wang"
date: "2023-02-11"
output: html_document
---

```{r read counts summary}
# module load goolf/7.1.0_3.1.4  R
# R
library(Rsamtools)
library(Rsubread)
bam_files <- c("/Users/clarawang/Downloads/DHT_1.sorted.bam", "/Users/clarawang/Downloads/DHT_2.sorted.bam","/Users/clarawang/Downloads/Veh_1.sorted.bam","/Users/clarawang/Downloads/Veh_2.sorted.bam")
gtf_file <- "/Users/clarawang/Downloads/gencode.v42.basic.annotation.gtf"
count_data <- featureCounts(files = bam_files, annot.ext = gtf_file, isGTFAnnotationFile = TRUE, GTF.featureType = "exon", GTF.attrType = "gene_name",isPairedEnd = TRUE)
write.table(count_table, file = "/scratch/jw4xtu/count.txt", sep = "\t", row.names = FALSE)
```

```{r Differential Expression Analysis}
library(DESeq2)
library(ggplot2)
library(dplyr)
unnorm <- read.table("/Users/clarawang/Desktop/counts_gene_name.txt", header = TRUE, row.names = "Geneid", sep = '\t', comment.char = "#", check.names = FALSE)
#delete the first five row which is useless
unnorm1 <- unnorm[,-c(1:5)]
#select rows that are significant
unnorm1 <- unnorm1[rowSums(unnorm1)>10, ]
unnorm1<-unnorm1%>%
  rename("DHT_1"="/Users/clarawang/Downloads/DHT_1.sorted.bam", "DHT_2"="/Users/clarawang/Downloads/DHT_2.sorted.bam", "Veh_1"="/Users/clarawang/Downloads/Veh_1.sorted.bam", "Veh_2"="/Users/clarawang/Downloads/Veh_2.sorted.bam")

#construct dds
samples <- data.frame(sampleID = c("DHT_1", "DHT_2", "Veh_1", "Veh_2"), sample = c("DHT", "DHT", "Veh", "Veh"))
row.names(samples) <- samples$sampleID
samples$sample <- factor(samples$sample, level = c("DHT", "Veh"))
counts <- as.matrix(unnorm1[row.names(samples)])
dds <- DESeqDataSetFromMatrix(countData = counts, colData = samples, design = ~ sample)
#analysis
dds <- DESeq(dds)
#see result and extract significant genes
AR_res <- results(dds, contrast = c("sample", "DHT", "Veh"), parallel = TRUE)
AR_0.01 <- AR_res[which(AR_res$padj<0.05), ]
print(AR_0.01)
write.csv(AR_0.01, file = "differential_expressed_gene1.csv")
#distinguish up-regulated or down-regulated genes
up_regulated <- AR_0.01[which(AR_0.01$log2FoldChange>1), ]
down_regulated <- AR_0.01[which(AR_0.01$log2FoldChange<(-1)), ]

write.csv(up_regulated, file = "up_regulated_gene1.csv")
write.csv(down_regulated, file = "down_regulated_gene1.csv")
```


```{r PCA}
#PCA
vsd <- vst(dds, blind = F)
plotPCA(vsd, intgroup = c("sample"))
plotPCA(vsd, intgroup = c("sample"), returnData = TRUE)
PCAreturn <- "
            PC1             PC2      group   sample    name
DHT_1	-4.829249	1.5695537	DHT	DHT	DHT_1
DHT_2	-1.542322	-2.6397634	DHT	DHT	DHT_2
Veh_1	3.001755	0.2662823	Veh	Veh	Veh_1
Veh_2	3.369816	0.8039274	Veh	Veh	Veh_2
"
PCAreturn <- read.table(header = TRUE, text = PCAreturn) 
View(PCAreturn)
PCAreturn$sample <- factor(PCAreturn$sample, levels = c("DHT", "Veh"))
ggplot(PCAreturn)+geom_point(aes(PC1,PC2, color = sample), size = 2)+labs(x = "PC1: 79% variance",y = "PC2: 17% variance", face = "bold")+theme_light(base_size = 15)
```


```{r FPKM}
counts_fpkm <- unnorm
for (clm in colnames(counts_fpkm)[6:9]) {
  col_fpkm=paste0(clm, "_FPKM")
  total=sum(counts_fpkm[clm])
  counts_fpkm[col_fpkm] = (counts_fpkm[clm]*10^6) / (counts_fpkm$Length*as.numeric(total)/1000)
}
write.csv(counts_fpkm, file = "FPKM.csv")
```

